src/sagas/backend.ts

```diff
@@ -277,6 +278,67 @@ function* backendSaga(): SagaIterator {
       handleResponseError(resp);
     }
   });
+
+  yield takeEvery(actionTypes.FETCH_NOTIFICATIONS, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+
+    if (!tokens.accessToken) {
+      return;
+    }
+
+    const notifications = yield getNotifications(tokens);
+
+    yield put(actions.updateNotifications(notifications));
+  });
+
+  yield takeEvery(actionTypes.ACKNOWLEDGE_NOTIFICATION, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+    const ids = (action as actionTypes.IAction).payload as number[];
+
+    if (ids.length === 0) {
+      return;
+    }
+
+    const notifications: Notification[] = yield select(
+      (state: IState) => state.session.notifications
+    );
+    const newNotifications: Notification[] = notifications.filter(
+      notification => !ids.includes(notification.id)
+    );
+
+    /* 
+      For responsiveness, we update the store's state first, then
+      check if the server rejects the acknowledgement and undo the changes if necessary
+    */
+
+    yield put(actions.updateNotifications(newNotifications));
+
+    const resp: Response | null = yield postAcknowledgeNotification(tokens, ids);
+
+    if (resp && resp.ok) {
+      return;
+    }
+
+    yield call(showWarningMessage, "Something went wrong, couldn't acknowledge");
+    yield put(actions.updateNotifications(notifications));
+  });
+
+  yield takeEvery(actionTypes.NOTIFY_CHATKIT_USERS, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+
+    const assessmentId = (action as actionTypes.IAction).payload.assessmentId || null;
```

Comment:
why did you choose to change it to null here? Why not leave it as `undefined`?

I don't think we strictly adhere to this but even the [Typescript Styleguide](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined) advocates `undefined` over `null`.